openadapt/app/dashboard/next.config.js

```diff
@@ -0,0 +1,30 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+    rewrites: async () => {
+        return [
+            {
+                source: '/api/:path*',
+                destination:
+                    process.env.NODE_ENV === 'development'
+                        ? 'http://127.0.0.1:8000/api/:path*'
+                        : '/api/',
+            },
+            {
+                source: '/docs',
+                destination:
+                    process.env.NODE_ENV === 'development'
+                        ? 'http://127.0.0.1:8000/docs'
+                        : '/api/docs',
+            },
+            {
+                source: '/openapi.json',
+                destination:
+                    process.env.NODE_ENV === 'development'
+                        ? 'http://127.0.0.1:8000/openapi.json'
+                        : '/api/openapi.json',
```

Comment:
Can we make this port configurable?

Ultimately we would like to unify the configuration with `openadapt.config`, but not necessary as part of this PR.

Reply:
Added `DASHBOARD_CLIENT_PORT` and `DASHBOARD_SERVER_PORT` to the config, which are then used to run the nextjs and fastapi servers. If any of these ports are being used at the time of launching the dashboard, it will display an error in the terminal.